enterprise/backend/src/metabase_enterprise/sso/api/sso.clj

```diff
@@ -68,16 +69,19 @@
 (api.macros/defendpoint :post "/logout"
   "Logout."
   [_route-params _query-params _body {cookies :cookies, :as _request}]
-  (let [metabase-session-id (get-in cookies [request/metabase-session-cookie :value])]
-    (api/check-exists? :model/Session metabase-session-id)
+  (let [metabase-session-key (get-in cookies [request/metabase-session-cookie :value])
+        metabase-session-key-hashed (session/hash-session-key metabase-session-key)]
+    (api/validate-session-key metabase-session-key)
```

Comment:
What if we just had a middleware that calls `validate-session-key` on every request if a key is present? This way if it's there, we can assume it's legit.

Reply:
That's a good idea. It has to be done on every request, so putting it there will make sure it's done once and only once